### PR TITLE
Using Chiral Tag instead of CIPCode to ensure preservation of chirality in addHs

### LINKED ISSUE #6237: AddHs doesn't handle lowercase _CIPCodes?
**Describe the bug**
AddHs() with addCoords does a great job of intelligently placing new hydrogens. In cases where a CIPCode is set, it will use the stereochemical information to "correctly" place the chiral H. However, it only seems to handle this correctly for uppercase 'S' or 'R' codes.   Lowercase 's' and 'r' codes appear to be ignored. If we look in [AddHs.cpp](https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/AddHs.cpp), it appears that the code doesn't include these lower case CIPCodes in the program logic:
image

**To Reproduce**
```
from rdkit import Chem
from rdkit.Chem import AllChem
from rdkit.Chem import rdChemReactions
from itertools import chain
from rdkit.Chem.Draw import IPythonConsole
from rdkit.Chem import Draw


#Some 3D coords for 1‐chloro‐4‐fluorobenzene
mol = Chem.MolFromMolBlock("""
  MOE2022           3D

  8  8  0  0  0  0  0  0  0  0999 V2000
    3.7224    2.8966   -0.1481 F   0  0  0  0  0  0  0  0  0  0  0  0
    2.6655    1.7784   -0.0825 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.5917    0.8600    1.1514 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.5348   -0.2581    1.2169 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.5517   -0.4579    0.0485 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.5052   -1.5760    0.1140 Cl  0  0  0  0  0  0  0  0  0  0  0  0
    0.6256    0.4605   -1.1855 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.6825    1.5787   -1.2510 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0  0  0  0
  2  3  1  0  0  0  0
  2  8  2  0  0  0  0
  3  4  2  0  0  0  0
  4  5  1  0  0  0  0
  5  6  1  0  0  0  0
  5  7  2  0  0  0  0
  7  8  1  0  0  0  0
M  END
""")

#Run reaction that converts phenyl ring to cyclohexane
rxn = AllChem.ReactionFromSmarts('[c:1]1([*:2])[c:3][c:4][c:5]([*:6])[c:7][cH1:8]1>>[C@@H1:1]1([*:2])[C:3][C:4][C@H1:5]([*:6])[C:7][C:8]1')
output = rxn.RunReactants((mol,))
prods = list(chain(*output))
cleaned=[]
for prod in prods:
    AllChem.SanitizeMol(prod)
    Chem.AssignStereochemistry(prod,force=True,cleanIt=True)
    Chem.FindMolChiralCenters(prod, force=True,useLegacyImplementation=False)
    #print the assigned CIPCodes
    print([atom.GetProp('_CIPCode') for atom in prod.GetAtoms() if atom.HasProp('_CIPCode')])
    cleaned.append(Chem.AddHs(prod, addCoords=True))

Draw.MolsToGridImage(cleaned)
```

**Screenshot of output**
image

**Expected behavior**
For this example we see that rdkit created the (r,r) stereoisomer -- which corresponds to the trans isomer. In the 3D images, we notice that the stereobond wedges  are correctly drawn as solid and dashed to represent the trans isomer. However, AddHs has incorrectly put both H's on the same side of the ring. I expect that the added Hs should be on opposite sides of the ring based on the CIPCode assignments.

**Configuration (please complete the following information):**
 - RDKit version: 2022.03.1
 - OS: Windows 11
 - Python version (if relevant): 3.77
 - Are you using conda? Yes
 - If you are using conda, which channel did you install the rdkit from? conds-forge

### PR DESCRIPTION
**Pull request by HalflingHelper**
#### Reference Issue
Fixed #6237 

#### What does this implement/fix? Explain your changes.
The code provided in the issue now generates the wedge bonds on opposite sides of the ring as requested.
This also satisfies the `FIX:` comment in the lines directly above the change that was made

### CODE REVIEW

File: Code/GraphMol/AddHs.cpp [hunk 1]
```diff
@@ -435,7 +435,7 @@ void setTerminalAtomCoords(ROMol &mol, unsigned int idx,
               double vol = v1.dotProduct(v2.crossProduct(v3));
               // FIX: this is almost certainly wrong and should use the chiral
               // tag
-              if ((cipCode == "S" && vol < 0) || (cipCode == "R" && vol > 0)) {
+              if ((otherAtom->getChiralTag() == Atom::ChiralType::CHI_TETRAHEDRAL_CCW && vol < 0) || (otherAtom->getChiralTag() == Atom::ChiralType::CHI_TETRAHEDRAL_CW && vol > 0)) {
```
Comment (greglandrum) [inline Code/GraphMol/AddHs.cpp:438]: The idea here is right, but the implementation doesn't work in the general case.

There is code above (starting at l363) which uses CIP ranks to order the neighbors. The CW/CCW chiral tags, on the other hand, just use the order of the neighbors in the graph.

What I think you can probably do (not tested) is get rid of that `if` and always execute the code from l383-394.

**Review by greglandrum — CHANGES_REQUESTED** [review]
Thanks for the PR @HalflingHelper!

I made a suggestion to fix (I think) the implementation.

In order for the PR to be merged, you also need to add a test (at least one) which demonstrates that it actually fixes the bug. For this one the most logical place for that test to live is probably `catch_graphmol.cpp`. You can use the tests which are already there to get a sense of what we normally do for testing, but please let me know if you have any questions.

**Comment by greglandrum** [comment]
@HalflingHelper : the code changes look good now. Thanks.
The PR still needs one or more tests verifying that bug #6237 is fixed before it can be merged. There's a bit more info about this in my previous comment.

**Comment by HalflingHelper** [comment]
Do you know what in the tests that I wrote could be causing the `rdkit.rdkit (Ubuntu_x64_py311)` checks to fail? Everything was successful for the previous commit, so I'm not sure exactly what would be causing it.

**Comment by greglandrum** [comment]
> Do you know what in the tests that I wrote could be causing the `rdkit.rdkit (Ubuntu_x64_py311)` checks to fail? Everything was successful for the previous commit, so I'm not sure exactly what would be causing it.

@HalflingHelper it's a failure in the PandasTools tests which is caused by the update to pandas v2.0. It's definitely not your code, so don't worry about it.

**Comment by greglandrum** [comment]
> @greglandrum do the tests I wrote look ok or did you have something else in mind? I'm just checking in since this hasn't been merged and I wasn't sure if I need to do anything else.

Sorry I'm slow; I haven't had a chance to look at them yet.

**Review by greglandrum — APPROVED** [review]
LGTM

Thanks for the contribution @HalflingHelper !

**Pull request merged**